From 5443af020352b2f8678fd8c1d505a66f27aa6777 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Fri, 14 Oct 2022 19:37:52 +0200
Subject: [PATCH] Mksquashfs: fix sysinfo.h missing on macos system

sysinfo.h and sysmacros.h is missing on MacOS. Exclude it from non
linux systems.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 squashfs-tools/mksquashfs.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
index 714ed35..da78ca7 100644
--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -49,9 +49,11 @@
 #include <sys/wait.h>
 #include <limits.h>
 #include <ctype.h>
-#include <sys/sysinfo.h>
 
-#ifndef linux
+#ifdef linux
+#include <sys/sysinfo.h>
+#include <sys/sysmacros.h>
+#else
 #include <sys/sysctl.h>
 #endif
 
-- 
2.37.2

